### ISSUE TITLE
Global Rate Limit

### ISSUE THREAD

**Issue by iamacarpet** [deranjer/goTorrent#9]
A global rate limit, to apply across all torrents like uTorrent has, would be great.

It would be good to have the option to set independently for upload / download, with an option to change on the fly without restart, if possible.

**Comment by deranjer** [comment]
As far as i can tell, https://github.com/anacrolix/torrent has support for a limiter, I just need to do the research and testing on exactly how to use it.  I'll also ask in gitter to see if anacrolix can support me on this.  If so, I could add an option to the settings.toml file that will allow you to set a global rate limit.  

If the underlying torrent library supports this I am optimistic on this being implemented in 0.2.0.

**Comment by deranjer** [comment]
Okay, doing some testing on the settings in the config.toml file as follows:
```
    #Limits your upload and download speed globally, all are averages and not burst protected (usually burst on start).
    #Low = ~.05MB/s, Medium = ~.5MB/s, High = ~1.5MB/s
    UploadRateLimit = "Low"  #Options are "Low", "Medium", "High", "Unlimited" #Unlimited is default
    DownloadRateLimit = "Low"
```

The speeds do appear to burst past the limits randomly but as far as I can tell it averages out.  This could also just be an issue with how I am measuring the speed of dl/upload.  Also not sure what I should set the levels at, let me know what you think.
